### PR TITLE
test properly usable_targets with different target types

### PR DESCRIPTION
**Pull request by TomasTomecek**
There was a bug internally and I just want to make it clear we test for the scenario which resulted into a failure.

### CODE REVIEW

File: colin/core/result.py [hunk 1]
```diff
@@ -56,6 +56,13 @@ class CheckResults(object):
     def __init__(self, results):
         self.results = CachingIterable(results)
 
+    @property
+    def results_per_check(self):
+        results_per_check = {}
+        for r in self.results:
+            results_per_check[r.check_name] = r
+        return results_per_check
```
Comment (jpopelka) [inline colin/core/result.py]: I know you'll say it looks like grc, but 

```python
    @property
    def results_per_check(self):
        return {r.check_name: r for r in self.results}
```

**Comment by jpopelka** [comment]
Is this expected ?

```
13:52:49 check_runner.py             35 DEBUG    Checking cmd_or_entrypoint
13:52:49 check_runner.py             41 WARNING  There was an error while performing check: Traceback (most recent call last):
13:52:49   File "/src/colin/core/check_runner.py", line 37, in _result_generator
13:52:49     yield check.check(target)
13:52:49   File "/src/colin/checks/best_practices.py", line 31, in check
13:52:49     raise RuntimeError("This check is not support now: skopeo doesn't provide this metadata.")
13:52:49 RuntimeError: This check is not support now: skopeo doesn't provide this metadata.
...
13:52:49 check_runner.py             35 DEBUG    Checking no_root
13:52:49 check_runner.py             41 WARNING  There was an error while performing check: Traceback (most recent call last):
13:52:49   File "/src/colin/core/check_runner.py", line 37, in _result_generator
13:52:49     yield check.check(target)
13:52:49   File "/src/colin/checks/best_practices.py", line 59, in check
13:52:49     raise RuntimeError("This check is not support now: skopeo doesn't provide this metadata.")
13:52:49 RuntimeError: This check is not support now: skopeo doesn't provide this metadata.
```

**Comment by lachmanfrantisek** [comment]
> Is this expected ?

Yes, it is expected.
https://github.com/user-cont/colin/blob/master/colin/checks/best_practices.py#L31

For ostree target, we have to wait for the skopeo support.
Now, I am working on the support for the podman images.

**Comment by jpopelka** [comment]
LGTM, +1 to commit messages :-)

**Comment by TomasTomecek** [comment]
let's ship it then? (Franto, sorry you'll need to rebase likely)

**Comment by lachmanfrantisek** [comment]
> (Franto, sorry you'll need to rebase likely)

No problem... At least, I have more tests.

**Pull request merged**